ArcGISRuntimeSDKQt_CppSamples/Analysis/ConfigureSubnetworkTrace/ConfigureSubnetworkTrace.cpp

```diff
@@ -0,0 +1,368 @@
+    }
+    else
+    {
+      m_textFieldVisible = true;
+    }
+    emit valueSelectionListModelChanged();
+    emit textFieldVisibleChanged();
+  }
+}
+
+void ConfigureSubnetworkTrace::addCondition(const QString& selectedAttribute, int selectedOperator, const QVariant& selectedValue)
+{
+  // NOTE: You may also create a UtilityCategoryComparison with UtilityNetworkDefinition.Categories and UtilityCategoryComparisonOperator.
+
+  if (!m_traceConfiguration)
+    m_traceConfiguration = new UtilityTraceConfiguration(this);
+
+  UtilityTraversability* traversability = m_traceConfiguration->traversability();
+  if (!traversability)
+    m_traceConfiguration->setTraversability(new UtilityTraversability(this));
+
+  UtilityNetworkAttribute* selectedNetworkAttribute = m_networkDefinition->networkAttribute(selectedAttribute);
+
+  const QVariant convertedSelectedValue = convertToDataType(selectedValue, selectedNetworkAttribute->dataType());
+
+  const UtilityAttributeComparisonOperator selectedOperatorEnum = static_cast<UtilityAttributeComparisonOperator>(selectedOperator);
+
+
+  // NOTE: You may also create a UtilityNetworkAttributeComparison with another NetworkAttribute.
+  UtilityTraceConditionalExpression* expression = new UtilityNetworkAttributeComparison(selectedNetworkAttribute, selectedOperatorEnum, convertedSelectedValue, this);
+
+  if (traversability->barriers())
+  {
+    UtilityTraceConditionalExpression* otherExpression = static_cast<UtilityTraceConditionalExpression*>(m_traceConfiguration->traversability()->barriers());
+
+    // NOTE: You may also combine expressions with UtilityTraceAndCondition
+    UtilityTraceConditionalExpression* combineExpressions = new UtilityTraceOrCondition(otherExpression, expression, this);
+
+    m_expressionBuilder = expressionToString(combineExpressions);
+    emit expressionBuilderChanged();
+
+    m_traceConfiguration->traversability()->setBarriers(combineExpressions);
+  }
+  else
+  {
+    m_expressionBuilder = expressionToString(expression);
+    emit expressionBuilderChanged();
+
+    m_traceConfiguration->traversability()->setBarriers(expression);
+  }
+}
+
+void ConfigureSubnetworkTrace::changeIncludeBarriersState(bool includeBarriers)
+{
+  m_traceConfiguration->setIncludeBarriers(includeBarriers);
+}
+
+void ConfigureSubnetworkTrace::changeIncludeContainersState(bool includeContainers)
+{
+  m_traceConfiguration->setIncludeContainers(includeContainers);
+}
+
+void ConfigureSubnetworkTrace::reset()
+{
+  // Reset the barrier condition to the initial value.
+  m_traceConfiguration->traversability()->setBarriers(m_initialExpression);
+  m_expressionBuilder.clear();
+  m_expressionBuilder = expressionToString(static_cast<UtilityTraceConditionalExpression*>(m_initialExpression));
+  emit expressionBuilderChanged();
+}
+
+void ConfigureSubnetworkTrace::trace()
+{
+  m_busy = true;
+  emit busyChanged();
+  if (!m_utilityNetwork || !m_utilityElementStartingLocation)
+  {
+    return;
```

Comment:
Is ` m_busy = true;` still correct if we do any early return here? I'm wondering if that will cause a progress spinner to active with nothing in progress.